sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -102,6 +102,27 @@ object functions {
     Column(literalExpr)
   }
 
+  /**
+   * Creates a [[Column]] of literal value.
+   *
+   * The passed in object is returned directly if it is already a [[Column]].
+   * If the object is a Scala Symbol, it is converted into a [[Column]] also.
+   * Otherwise, a new [[Column]] is created to represent the literal value.
+   *
+   * @group normal_funcs
+   * @since 2.2.0
+   */
+  def lit2[T : TypeTag](literal: T): Column = {
```

Comment:
Do you think there is a way we can actually avoid this?

If we must  why not name it `typedLit`?

cc @cloud-fan